File: Objc/Search/BinarySearch/BinarySearch.m
//
//  BinarySearch.m
//  Algorithm
//
//  Created by Kuan-Wei Lin on 3/8/17.
//  Copyright © 2017 TaiwanMobile. All rights reserved.
//

#import "BinarySearch.h"

@implementation BinarySearch

- (void)binarySearch:(NSArray *)array key:(NSNumber *)key
{
    NSMutableArray *data = [array mutableCopy];
    
    NSUInteger left = 0;
    NSUInteger right = data.count - 1;
    NSUInteger middele = 0;
    
    while (left <= right)
    {
        middele = (left + right) / 2;
        
        if (data[middele] == key)
        {
            NSLog(@"result = %lu", middele);
            return;
        }
        else if (data[middele] < key)
        {
            left = middele + 1;
        }
        else
        {
            right = middele - 1;
        }
    }
    
    NSLog(@"-1 not founded");
}

@end

File: Swift/Search/BinarySearch.swift
//
//  BinarySearch.swift
//  Algorithm_Swift
//
//  Created by Kuan-Wei Lin on 3/16/17.
//  Copyright © 2017 cracktheterm. All rights reserved.
//

import UIKit

class BinarySearch: NSObject {

    func binarySearch(array:[Int], key:Int)
    {
        var left = 0;
        var right = array.count - 1;
        var middle = 0;
        
        while left <= right
        {
            middle = (left + right) / 2;
            
            if array[middle] == key
            {
                print("result = \(middle)");
                return;
            }
            else if array[middle] < key
            {
                left = middle + 1;
            }
            else
            {
                right = middle - 1;
            }
        }
        
        print("-1 not founded")
    }
}
